droidlet/dashboard/web/src/StateManager.js

```diff
@@ -472,6 +495,30 @@ class StateManager {
         });
       }
     });
+
+    // Update feed state here because it's received first
+    if (!this.curFeedState.pose) {
+      this.curFeedState.pose = {
+        x: res.x, 
+        y: res.y, 
+        yaw: res.yaw,
+      }
+    } else if (res && 
+      (res.x !== this.curFeedState.pose.x || 
+      res.y !== this.curFeedState.pose.y || 
+      res.yaw !== this.curFeedState.pose.yaw)) {
+      this.prevFeedState = this.curFeedState
+      this.curFeedState = {
+        rgbImg: null, 
+        depthOrg: null, 
+        masks: null, 
+        pose: {
+          x: res.x, 
+          y: res.y, 
+          yaw: res.yaw, 
```

Comment:
Why does rgbImg, depthOrg, and masks all being reset to null? wouldn't that override previous data?

Reply:
This should be solved with the most recent commit. The only time the properties can be null should only be at the start.